pennylane/templates/tensornetworks/ttn.py

```diff
@@ -0,0 +1,210 @@
+                    x
+                    + 2 ** (j - 1) * n_block_wires // 2
+                    + 2 ** (j - 1) * n_block_wires // 2
+                    - n_block_wires // 2,
+                    x
+                    + 2 ** (j - 1) * n_block_wires // 2
+                    + n_block_wires // 2
+                    + 2 ** (j - 1) * n_block_wires // 2
+                    - n_block_wires // 2,
+                )
+            ]
+            for j in range(1, n_layers + 1)
+            for x in range(0, n_wires - n_block_wires // 2, 2 ** (j - 1) * n_block_wires)
+        ]
+    )
+
+    return layers
+
+
+class TTN(Operation):
+    """The TTN template broadcasts an input circuit across many wires following the architecture of a tree tensor network.
+    The result is similar to the architecture in `arXiv:1803.11537 <https://arxiv.org/abs/1803.11537>`_.
+
+    The argument ``block`` is a user-defined quantum circuit.``block`` should have two arguments: ``weights`` and ``wires``.
+    For clarity, it is recommended to use a one-dimensional list or array for the block weights.
+
+    Args:
+        wires (Iterable): wires that the template acts on
+        n_block_wires (int): number of wires per block
+        block (Callable): quantum circuit that defines a block
+        n_params_block (int): the number of parameters in a block; equal to the length of the ``weights`` argument in ``block``
+        template_weights (Sequence): list containing the weights for all blocks
+
+    .. note::
+
+        The expected number of blocks can be obtained from ``qml.TTN.n_blocks(wires, n_block_wires)``.
+        The length of ``template_weights`` argument should match the number of blocks.
+
+    .. UsageDetails::
+
+        This example demonstrates the use of ``TTN`` for a simple block.
+
+        .. code-block:: python
+
+            import pennylane as qml
+            import numpy as np
+
+            def block(weights, wires):
+                qml.CNOT(wires=[wires[0],wires[1]])
+                qml.RY(weights[0], wires=wires[0])
+                qml.RY(weights[1], wires=wires[1])
+
+            n_wires = 4
+            n_block_wires = 2
+            n_params_block = 2
+            n_blocks = qml.TTN.get_n_blocks(range(n_wires),n_block_wires)
+            template_weights = [[0.1,-0.3]]*n_blocks
+
+            dev= qml.device('default.qubit',wires=range(n_wires))
+            @qml.qnode(dev)
+            def circuit(template_weights):
+                qml.TTN(range(n_wires),n_block_wires,block, n_params_block, template_weights)
+                return qml.expval(qml.PauliZ(wires=n_wires-1))
+
+        >>> print(qml.draw(circuit,expansion_strategy='device')(template_weights))
+        0: ──╭C──RY(0.1)─────────────────┤
+        1: ──╰X──RY(-0.3)──╭C──RY(0.1)───┤
+        2: ──╭C──RY(0.1)───│─────────────┤
+        3: ──╰X──RY(-0.3)──╰X──RY(-0.3)──┤ ⟨Z⟩
+    """
+
+    num_params = 1
+    num_wires = AnyWires
+    par_domain = "A"
```

Comment:
We removed this redundant class attribute a while back, can you try to delete the line and see what happens?